### ISSUE TITLE
reload causes metrics disappear in K8S

### ISSUE THREAD

**Issue by ckcd** [coredns/coredns#2111]
Hi,

I'm using CoreDNS in kubernetes 1.9, the "CoreDNS" pod mount a configMap which describe the "Corefile". 
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: coredns
  namespace: kube-system
data:
  Corefile: |
    .:53 {
        errors
        health
        kubernetes cluster.local in-addr.arpa ip6.arpa {
          pods insecure
          upstream
          fallthrough in-addr.arpa ip6.arpa
        }
        prometheus :9153
        proxy . 114.114.114.114:53 119.29.29.29:53
        cache
        loop
        reload
    }
```

And I use curl to get metrics:
```
curl 192.168.164.4:9153/metrics
```

When I modify the configmap,  changes "114.114.114.114" to "223.5.5.5", and after 1 minutes I check the pod log, we can see the **reload**.
```
root@deploy:~/changkun/monitor# kubectl -n kube-system logs coredns-7964869f69-p27g5
.:53
2018/09/20 10:10:41 [INFO] CoreDNS-1.2.2
2018/09/20 10:10:41 [INFO] linux/amd64, go1.11, eb51e8b
CoreDNS-1.2.2
linux/amd64, go1.11, eb51e8b
2018/09/20 10:10:41 [INFO] plugin/reload: Running configuration MD5 = d52aa6c116a9f151a86868b56cbd1ea3
2018/09/20 10:13:33 [INFO] Reloading
2018/09/20 10:13:33 [INFO] plugin/reload: Running configuration MD5 = 679c78fa09bab0ffe9bc6887af278c49
2018/09/20 10:13:33 [INFO] Reloading complete
```
But, after the update, the metrics is incomplete, I **cannot** see any metrics of any plugins, such as "coredns_cache_misses_total" and "coredns_proxy_request_count_total".  Only has part of metrics, such as "coredns_dns_request_type_count_total"
```
# HELP coredns_build_info A metric with a constant '1' value labeled by version, revision, and goversion from which CoreDNS was built.
# TYPE coredns_build_info gauge
coredns_build_info{goversion="go1.11",revision="eb51e8b",version="1.2.2"} 1
# HELP coredns_dns_request_count_total Counter of DNS requests made per zone, protocol and family.
# TYPE coredns_dns_request_count_total counter
coredns_dns_request_count_total{family="1",proto="udp",server="dns://:53",zone="."} 78581
# HELP coredns_dns_request_duration_seconds Histogram of the time (in seconds) each request took.
# TYPE coredns_dns_request_duration_seconds histogram
coredns_dns_request_duration_seconds_bucket{server="dns://:53",zone=".",le="0.00025"} 78336

...

```

Seems like, after a reload, all metrics of all plugins will be disappear, but the metrics of main functions will be remain.

Is there any body has this problem? or is there any mistakes in my methods?

**Comment by fturib** [comment]
That issue was observed here : #2079 (after CoreDNS 1.2.2 was released)
It is now fixed with #2080.

**Comment by ckcd** [comment]
@fturib Thanks a lot, I build the latest code and the problem is solved~ Thks!!!
